langchain/vectorstores/azuresearch.py

```diff
@@ -0,0 +1,484 @@
+"""Wrapper around Azure Cognitive Search."""
+from __future__ import annotations
+
+import json
+import logging
+import os
+import uuid
+from typing import Any, Callable, Dict, Iterable, List, Optional, Tuple, Type
+
+import numpy as np
+from azure.core.credentials import AzureKeyCredential
+from azure.core.exceptions import ResourceNotFoundError
+from azure.identity import DefaultAzureCredential
+from azure.search.documents import SearchClient
+from azure.search.documents.indexes import SearchIndexClient
+from azure.search.documents.indexes.models import (
+    PrioritizedFields,
+    SearchableField,
+    SearchField,
+    SearchFieldDataType,
+    SearchIndex,
+    SemanticConfiguration,
+    SemanticField,
+    SemanticSettings,
+    SimpleField,
+    VectorSearch,
+    VectorSearchAlgorithmConfiguration,
+)
+from azure.search.documents.models import Vector
+from pydantic import BaseModel, root_validator
+
+from langchain.docstore.document import Document
+from langchain.embeddings.base import Embeddings
+from langchain.schema import BaseRetriever
+from langchain.vectorstores.base import VectorStore
+
+logger = logging.getLogger()
+
+AZURESEARCH_DIMENSIONS = int(
+    os.environ.get("AZURESEARCH_DIMENSIONS", 1536)
+)  # Default to OpenAI's ada-002 embedding model vector size
+
+# Allow overriding field names for Azure Search
+FIELDS_ID = os.environ.get("AZURESEARCH_FIELDS_ID", "id")
+FIELDS_TITLE = os.environ.get("AZURESEARCH_FIELDS_TITLE", "title")
+FIELDS_CONTENT = os.environ.get("AZURESEARCH_FIELDS_CONTENT", "content")
+FIELDS_CONTENT_VECTOR = os.environ.get(
+    "AZURESEARCH_FIELDS_CONTENT_VECTOR", "content_vector"
+)
+FIELDS_TAG = os.environ.get("AZURESEARCH_FIELDS_TAG", "tag")
+FIELDS_METADATA = os.environ.get("AZURESEARCH_FIELDS_TAG", "metadata")
+
+MAX_UPLOAD_BATCH_SIZE = 1000
+
+
+def get_search_client(
+    endpoint: str,
+    key: str,
+    index_name: str,
+    semantic_configuration_name: Optional[str] = None,
+) -> SearchClient:
+    if key is None:
+        credential = DefaultAzureCredential()
+    else:
+        credential = AzureKeyCredential(key)
+    index_client: SearchIndexClient = SearchIndexClient(
+        endpoint=endpoint, credential=credential
+    )
+    try:
+        index_client.get_index(name=index_name)
+    except ResourceNotFoundError:
+        # Fields configuration
+        fields = [
```

Comment:
I was wondering this as well, it seems tag and title are left empty in almost all use cases.  I assumed this was driven by the ACS pattern defining prioritized fields and maybe they aren't useful as a langchain vector store.  I think it would be nice if these fields could get filled in during insertion though, maybe with a user specified LLM to generate them, and use them as another way to find the closest match...

Reply:
As discussed offline, will provide another example with a configurable Index.